ledger/src/blockstore_db.rs

```diff
@@ -305,8 +307,40 @@ impl Rocks {
             new_cf_descriptor::<BankHash>(&access_type, &oldest_slot),
             new_cf_descriptor::<Root>(&access_type, &oldest_slot),
             new_cf_descriptor::<Index>(&access_type, &oldest_slot),
-            new_cf_descriptor::<ShredData>(&access_type, &oldest_slot),
-            new_cf_descriptor::<ShredCode>(&access_type, &oldest_slot),
+            match options.shred_storage_type {
```

Comment:
Out of curiosity, any potential issues if a validator opens a ledger in one compaction type, then shuts down and opens the ledger with the other compaction type?

Reply:
Good question.  It's actually covered in the proposal #21358.

From Level to FIFO is simple, because every single file under Level compaction has a disjoint key range, which makes it a valid FIFO format.

From FIFO to Level, the DB will hit the write stall immediately because there are too many L0 files.  Typically running manual compaction first before opening the DB is suggested.

Reply:
Hmmm ok I think we need to think about how to expose the details of that second case to operators.

Is there a way to tell that the db was formerly opened in FIFO and is now switching to level so we can notify the operator they should perform manual compaction?

Reply:
> Is there a way to tell that the db was formerly opened in FIFO and is now switching to level so we can notify the operator they should perform manual compaction?

I remember RocksDB itself does not store its options.  The usual way to do it is to store the options inside an options file https://github.com/facebook/rocksdb/wiki/RocksDB-Options-File.  Eventually, we might want to use RocksDB-Options-file to manage RocksDB options.

For the FIFO change, what we can do is to first have the FIFO code landed, then another PR to make it configurable via the command line arguments.  Then we stay at this point for a while to make sure we will never revert back to a release where we cannot configure whether we want to use FIFO or Level.  Once we reach that point, we can safely assume even when we roll back.  The validator will still understand the command line argument and correctly use FIFO or Level after the revert.